crates/storage/provider/src/providers/blockchain_provider.rs

```diff
@@ -409,10 +655,41 @@ where
     }
 
     fn receipt_by_hash(&self, hash: TxHash) -> ProviderResult<Option<Receipt>> {
+        for block_state in self.canonical_in_memory_state.canonical_chain() {
+            let executed_block = block_state.block();
+            let block = executed_block.block();
+            let receipts = block_state.executed_block_receipts();
+
+            // assuming 1:1 correspondence between transactions and receipts
```

Comment:
is this always true, do we have one receipt per tx?

Reply:
yeah if you look at the tables they're both keyed by number and the only junction table is the one that maps txhash to number